CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# Minimum CMake required
+cmake_minimum_required(VERSION 3.1)
+
+# Project
+project(onnxir C CXX)
+
+# Set C++11 as standard for the whole project
+set(CMAKE_CXX_STANDARD 11)
+
+set(ONNXIR_ROOT ${PROJECT_SOURCE_DIR})
+if (WIN32)
+    # parallel build
+    add_compile_options(/MP)
+    add_compile_options(/WX)
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+    set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
+endif()
+add_subdirectory(third_party/pybind11)
+FIND_PACKAGE(Protobuf REQUIRED)
+include_directories(
+    ${ONNXIR_ROOT}
+    ${CMAKE_CURRENT_BINARY_DIR}
+    # External dependencies.
+    ${PROTOBUF_INCLUDE_DIRS}
+    ${PYBIND11_INCLUDE_DIR}
+    ${PYTHON_INCLUDE_DIRS}
+)
+
+PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS
+    ${ONNXIR_ROOT}/onnx/onnx.proto
+    ${ONNXIR_ROOT}/onnx/onnx-operators.proto
+)
+
+file(GLOB_RECURSE onnx_src
+    "${ONNXIR_ROOT}/onnx/*.h"
+    "${ONNXIR_ROOT}/onnx/*.cc"
+)
+
+file(GLOB_RECURSE onnx_exclude_src
+    "${ONNXIR_ROOT}/onnx/onnx.pb.h"
+    "${ONNXIR_ROOT}/onnx/onnx.pb.cc"
+    "${ONNXIR_ROOT}/onnx/onnx-operators.pb.h"
+    "${ONNXIR_ROOT}/onnx/onnx-operators.pb.cc"
+)
+
+# Temp solution to remove files generated by calling 'pip install ..."
+# Should be removed after changing setup.py to use cmake to build c++ part.
+list(LENGTH onnx_exclude_src list_len)
+if (list_len GREATER 1)
+    list(REMOVE_ITEM onnx_src ${onnx_exclude_src})
+endif()
+
+add_library(onnxir ${PROTO_SRCS} ${PROTO_HDRS} ${onnx_src})
```

Comment:
In theory, onnxir should also depend on the protobuf libraries (if we build it as libonnxir.so). This might only work for cases where BUILD_SHARED_LIBS=OFF.

Reply:
that should be added as a link dependency in application-layer cmake file, right?

Reply:
It's better if you declare it in the cmake. Then if someone else recursively includes this cmake, and they depend on onnxir, they will automatically see the protobuf dependency. If you omit it here they won't.